modules/navier_stokes/src/executioners/SIMPLE.C

```diff
@@ -0,0 +1,539 @@
+
+    // Only evaluating right hand side which is R(0)
+    _problem.computeResidual(*zero_solution, rhs, _momentum_system_numbers[system_i]);
+    // Sadly, this returns -b so we multiply with -1
+    rhs.scale(-1.0);
+
+    // Still need to relax the right hand side with the same vector
+    relaxRightHandSide(rhs, solution, *diff_diagonal);
+
+    // The normalization factor depends on the right hand side so we need to recompute it for this
+    // component
+    norm_factor = computeNormalizationFactor(solution, mmat, rhs);
+
+    // Solve this component. We don't update the ghosted solution yet, that will come at the end of
+    // the corrector step
+    momentum_solver.solve(mmat, mmat, solution, rhs, 1e-10, 100);
+    // Save the normalized residual
+    normalized_residuals.push_back(momentum_solver.get_initial_residual() / norm_factor);
+
+    if (_print_fields)
+    {
+      _console << " rhs when we solve " << std::endl;
+      rhs.print();
+      _console << " velocity solution component " << system_i << std::endl;
+      solution.print();
+      _console << "Norm factor " << norm_factor << std::endl;
+      _console << Moose::stringify(momentum_solver.get_initial_residual()) << std::endl;
+    }
+  }
+
+  return normalized_residuals;
+}
+
+Real
+SIMPLE::solvePressureCorrector()
+{
+  _problem.setCurrentNonlinearSystem(_pressure_sys_number);
+
+  // We will need some members from the implocot nonlinear system
+  NonlinearImplicitSystem & pressure_system =
+      dynamic_cast<NonlinearImplicitSystem &>(_pressure_system.system());
+
+  // We will need the solution, the right hand side and the matrix
+  NumericVector<Number> & current_local_solution = *(pressure_system.current_local_solution);
+  NumericVector<Number> & solution = *(pressure_system.solution);
+  SparseMatrix<Number> & mmat = *(pressure_system.matrix);
+  NumericVector<Number> & rhs = *(pressure_system.rhs);
+
+  // Fetch the linear solver from the system
+  PetscLinearSolver<Real> & pressure_solver =
+      dynamic_cast<PetscLinearSolver<Real> &>(*pressure_system.get_linear_solver());
+
+  // We need a zero vector to be able to emulate the Ax=b system by evaluating the
+  // residual and jacobian. Unfortunately, this will leave us with the -b on the righ hand side
+  // so we correct it by multiplying it with (-1)
+  auto zero_solution = current_local_solution.zero_clone();
+  _problem.computeResidualAndJacobian(*zero_solution, rhs, mmat);
+  rhs.scale(-1.0);
+
+  if (_print_fields)
+  {
+    _console << "Pressure matrix" << std::endl;
+    mmat.print();
+  }
+
+  // We compute the normalization factors based on the fluxes
+  Real norm_factor = computeNormalizationFactor(solution, mmat, rhs);
+
+  // We need the non-preconditioned norm to be consistent with the norm factor
+  KSPSetNormType(pressure_solver.ksp(), KSP_NORM_UNPRECONDITIONED);
+
+  // Solve the system and update current local solution
+  pressure_solver.solve(mmat, mmat, solution, rhs, 1e-10, 100);
```

Comment:
For temporary arguments I often like to do something like `/*tol=*/1e-10, /*max_its=*/100` to convey intent. Not critical though

Reply:
(I'm guessing what those arguments mean)